solidity/test/isms/ExternalBridgeTest.sol

```diff
@@ -0,0 +1,272 @@
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(hook),
+            address(ism),
+            0,
+            messageId
+        );
+
+        ism.verify(externalCalldata, encodedMessage);
+        assertTrue(ism.isVerified(encodedMessage));
+    }
+
+    /* ============ ISM.verify ============ */
+
+    function test_verify_revertWhen_invalidMetadata() public virtual {
+        bool isValid;
+        try ism.verify(new bytes(0), encodedMessage) returns (bool _isValid) {
+            isValid = _isValid;
+        } catch {}
+        assertFalse(isValid);
+    }
+
+    function test_verify_msgValue_asyncCall() public virtual {
+        bytes memory encodedHookData = _encodeHookData(messageId);
+        _externalBridgeDestinationCall(encodedHookData, 1 ether);
+
+        assertTrue(ism.verify(new bytes(0), encodedMessage));
+        assertEq(address(testRecipient).balance, 1 ether);
+    }
+
+    function test_verify_msgValue_externalBridgeCall() public virtual {
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(hook),
+            address(ism),
+            1 ether,
+            messageId
+        );
+        ism.verify(externalCalldata, encodedMessage);
+        assertEq(address(testRecipient).balance, 1 ether);
+    }
+
+    function test_verify_revertsWhen_invalidIsm() public virtual {
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(hook),
+            address(this),
+            0,
+            messageId
+        );
+
+        vm.expectRevert();
+        assertFalse(ism.verify(externalCalldata, encodedMessage));
+    }
+
+    function test_verify_revertsWhen_notAuthorizedHook() public virtual {
+        _setExternalOriginSender(address(this));
+
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(this),
+            address(ism),
+            0,
+            messageId
+        );
+
+        // external call
+        vm.expectRevert(unauthorizedHookError);
+        assertFalse(ism.verify(externalCalldata, encodedMessage));
+
+        // async call vm.expectRevert(NotCrossChainCall.selector);
+        vm.expectRevert();
+        _externalBridgeDestinationCall(externalCalldata, 0);
+        assertFalse(ism.isVerified(encodedMessage));
+    }
+
+    function test_verify_revertsWhen_incorrectMessageId() public virtual {
+        bytes32 incorrectMessageId = keccak256("incorrect message id");
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(hook),
+            address(ism),
+            0,
+            incorrectMessageId
+        );
+
+        // external call
+        bool isValid;
+        try ism.verify(externalCalldata, encodedMessage) returns (
+            bool _isValid
+        ) {
+            isValid = _isValid;
+        } catch {}
+        assertFalse(isValid);
```

Comment:
ditto here, expect revert?

Reply:
same as before, verify may not revert and just return false.

Reply:
fixed by overridding case by case